packages/transaction_signing_gateway/lib/wallet_derivator.dart

```diff
@@ -0,0 +1,12 @@
+import 'package:dartz/dartz.dart';
+import 'package:transaction_signing_gateway/model/network_info.dart';
+import 'package:transaction_signing_gateway/model/private_wallet_credentials.dart';
+import 'package:transaction_signing_gateway/model/wallet_derivation_failure.dart';
+
+abstract class WalletDerivator {
```

Comment:
We would also need to add a `derive` method inside the `TransactionSigningGateway` class and make it work similar to what we have for transaction signing, where we have a list of derivators and look for one which is capable of handling the derivation

Reply:
✅